plugins/janus_streaming.c

```diff
@@ -6519,6 +6533,23 @@ static int janus_streaming_rtsp_connect_to_server(janus_streaming_mountpoint *mp
 	}
 	long code = 0;
 	res = curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &code);
+#if CURL_AT_LEAST_VERSION(7, 62, 0)
+	if(source->rtsp_quirk && code == 404) {
+		/*Possibly a quirk in the RTSP server, where the DESCRIBE request expects a path only.*/
+		CURLU *curl_u = curl_url();
+		char *path = NULL;
+		if(!(curl_url_set(curl_u, CURLUPART_URL, source->rtsp_url, 0))) {
+			if(!(curl_url_get(curl_u, CURLUPART_PATH, &path, 0))) {
+				curl_easy_setopt(curl, CURLOPT_RTSP_STREAM_URI, path);
+				res = curl_easy_perform(curl);
+				source->rtsp_stream_uri = g_strdup(path);
```

Comment:
Shouldn't we check what `curl_easy_perform` and/or `curl_easy_getinfo` returned, before saving this property? If we got a `404` or a different error again, then the quirk didn't help. Besides, `path` may still be `NULL` maybe?

Reply:
I don't think path can be null. The curl documentation seems to indicate that it will be set to "/" at the worst. (https://curl.se/libcurl/c/curl_url_get.html)